src/Manager.sol

```diff
@@ -405,79 +401,78 @@ abstract contract Manager is
 
-                // query own token balance after transfer
-                uint256 balanceAfter = getTokenBalanceOf(_token, address(this));
+            // query own token balance after transfer
+            uint256 balanceAfter = getTokenBalanceOf(_token, address(this));
 
-                // correct amount for potential transfer fees
-                amount = balanceAfter - balanceBefore;
-            } else if (_mode == Mode.BURNING) {
-                // query sender's token balance before transfer
-                uint256 balanceBefore = getTokenBalanceOf(_token, msg.sender);
+            // correct amount for potential transfer fees
+            amount = balanceAfter - balanceBefore;
+        } else if (_mode == Mode.BURNING) {
+            // query sender's token balance before transfer
+            uint256 balanceBefore = getTokenBalanceOf(_token, msg.sender);
 
-                // call the token's burn function to burn the sender's token
-                ERC20Burnable(_token).burnFrom(msg.sender, amount);
+            // call the token's burn function to burn the sender's token
+            ERC20Burnable(_token).burnFrom(msg.sender, amount);
 
-                // query sender's token balance after transfer
-                uint256 balanceAfter = getTokenBalanceOf(_token, msg.sender);
+            // query sender's token balance after transfer
+            uint256 balanceAfter = getTokenBalanceOf(_token, msg.sender);
 
-                // correct amount for potential burn fees
-                amount = balanceAfter - balanceBefore;
-            } else {
-                revert InvalidMode(uint8(_mode));
-            }
+            // correct amount for potential burn fees
+            amount = balanceAfter - balanceBefore;
+        } else {
+            revert InvalidMode(uint8(_mode));
+        }
+
+        // now check rate limits
+        bool isAmountRateLimited = _isOutboundAmountRateLimited(amount);
+        if (shouldQueue && isAmountRateLimited) {
+            // queue up and return
+            uint64 queueSequence = _enqueueOutboundTransfer(amount, recipientChain, recipient);
 
-            // now check rate limits
-            bool isAmountRateLimited = _isOutboundAmountRateLimited(amount);
-            if (shouldQueue && isAmountRateLimited) {
-                // queue up and return
-                uint64 queueSequence = _enqueueOutboundTransfer(amount, recipientChain, recipient);
+            // refund the price quote back to sender
+            payable(msg.sender).transfer(msg.value);
 
-                // refund the price quote back to sender
-                payable(msg.sender).transfer(msg.value);
+            // return the sequence in the queue
+            return queueSequence;
+        }
 
-                // return the sequence in the queue
-                return queueSequence;
-            }
+        // otherwise, consume the outbound amount
+        _consumeOutboundAmount(amount);
```

Comment:
this may be stylistic but since the `false` code path for `shouldQueue` reverts, imo makes more sense to check that condition first and then proceed to the enqueuing logic

Reply:
This is technically checked up front before the entire if-statement with
```
bool isAmountRateLimited = _isOutboundAmountRateLimited(amount);
```

We then only proceed to the enqueuing logic if that check passes. I would say this is more stylistic but I'm curious to get @djb15 's thoughts on whether one structuring is more readable than the other.

Reply:
yea logically i think its fine. more of a question of how structure early returns in the control flow

Reply:
As I was reviewing this PR I actually got confused by this and thought there was a bug, so I agree with Rahul here. Can we have an explicit `if (!shouldQueue && isAmountRateLimited)` before the queuing logic or something similar?